### PR TITLE
drivers: sensor: Lis2dh data reading fix

### PR DESCRIPTION
**Pull request by ekalugin**
For some reason I2C burst mode in lis2dh works unreliably and just fills the buffer with continuously repeating first byte. Reading bytes one-by-one solves the problem.
Also a compilation error is fixed in header file.

### CODE REVIEW

**Comment by erwango** [comment]
@avisconti

File: drivers/sensor/lis2dh/lis2dh.c [hunk 1]
```diff
@@ -121,18 +121,19 @@ static int lis2dh_sample_fetch(struct device *dev, enum sensor_channel chan)
...
-				   lis2dh->sample.raw,
-				   sizeof(lis2dh->sample.raw));
+	for(i = 0; i < sizeof(lis2dh->sample.raw); i++) {
+		status |= lis2dh_reg_read_byte(lis2dh->bus,
+			LIS2DH_REG_STATUS + i, &lis2dh->sample.raw[i]);
+	}
+
 	if (status < 0) {
 		SYS_LOG_WRN("Could not read accel axis data");
 		return status;
```
Comment (avisconti) [inline drivers/sensor/lis2dh/lis2dh.c]: Actually I think that the sensor works fine with I2C burst transaction.
What you have to do is simply set the I2C AUTOINCREMENT bit, like the lis3dh driver does. If you see here in this driver the bit is defined (LIS2DH_AUTOINCREMENT_ADDR) but is not used. Take a look to lis3dh driver and of course to the lis2dh datasheet.

Comment (avisconti) [reply drivers/sensor/lis2dh/lis2dh.c]: @ekalugin 
You have to do something like:
status = lis2dh_burst_read(dev, LIS2DH_REG_STATUS | LIS2DH_AUTOINCREMENT_ADDR, .....

Comment (avisconti) [reply drivers/sensor/lis2dh/lis2dh.c]: Pls let me know if it works

Comment (avisconti) [reply drivers/sensor/lis2dh/lis2dh.c]: I report the LIS2DH datasheet section that explains the I2C auto-increment address:

The I2C embedded inside the LIS2DH behaves like a slave device and the following protocol must be adhered to. After the start condition (ST) a slave address is sent, once a slave acknowledge (SAK) has been returned, a 8-bit sub-address (SUB) is transmitted: the 7 LSb represent the actual register address while the MSB enables address auto increment. If the MSb of the SUB field is ‘1’, the SUB (register address) is automatically increased to allow multiple data read/write.

**Comment by avisconti** [comment]
> Yep, I've just tried to add LIS2DH_AUTOINCREMENT_ADDR, and it didn't help, unfortunately.

Actually it is strange. I have no time to test it here on my side as I'm doing other things. But I expect it to work on LIS2DH. Are you saying that  you always read the value at the same address (i.e the LIS2DH_REG_STATUS)?

**Comment by avisconti** [comment]
@ekalugin 
Can you send to me the code you are trying?

**Comment by ekalugin** [comment]
@avisconti I wonder how to send it. 
Try to access this gist: https://gist.github.com/ekalugin/1509e6fa0eb80416dc18b7f66c0138ef

**Comment by avisconti** [comment]
@galak @ekalugin 
In my opinion the #8264 should have fixed already what this PR was attempting to.

**Pull request closed without merging**